processor/src/test/java/org/mapstruct/ap/test/bugs/_1826/Issue1826Mapper.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._1826;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.NullValueCheckStrategy;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS)
```

Comment:
I think you need to add another test case (and possibly some additional coding) to test whether the `nullValueCheckStrategy` actually works.

Reply:
To be honest, I dont think the NullValueCheckStrategy property effects this code.  The source that gets generated here is identical without / without the strategy defined, so i can remove this from the test mapper definition if that helps.   I don't see any switches in the template that change the logic based on the  NullValueCheckStrategy setting.  What exactly would i be testing here?

Reply:
I think that you should control the addition of the null check, having a possibility to opt-in. That would be  in line with our documentation. 

> I don't see any switches in the template that change the logic based on the NullValueCheckStrategy setting. What exactly would i be testing here?

I think that should be added to stay in line with the documentation and other places... It is present in the `PropertyMapping` and then `NullValueCheck` can be added to the creation of the backing bean of the template. BTW: this being a non update method it is the `NullValueCheckStrategy` you're after... not the `NullValuePropertyCheckStrategy`.

Reply:
Hold on before you implement this.. Perhaps you're right.. I added also some comment to the issue on this.

Reply:
@sjaakd I am not sure that we need something special for this. This is only happening for extracting nested source value, if we add another option it might be overcomplicating it 😄

Reply:
👍 ok.. Agreed.

Reply:
@jakraska then my only remaining comment is to remove argument of the `@Mapper`..

Reply:
@sjaakd will do

Reply:
The extra parameters have been removed from the @mapper annotation.